### ISSUE TITLE
Rewards link changed

### ISSUE THREAD

**Issue by wearysky** [h4x0rm1k3/ABS#9]
Too lazy to branch and commit separately, but just an FYI @h4x0rm1k3 it looks like Microsoft has fully committed to redirecting all rewards.microsoft.com browsing sessions to rewards.bing.com some time this week. Might not be rolled out to everyone, but it's definitely happening for me now. I had to update all references to the old URL to point to the new one, otherwise it wasn't auto-clicking the links anymore.

**Comment by wearysky** [comment]
For those who want to manually fix this on their own until a commit goes in - it's manifest.json and constants.js that need editing.

**Comment by ChippewaBarr** [comment]
> For those who want to manually fix this on their own until a commit goes in - it's manifest.json and constants.js that need editing.

For the edits, do you mean simply change all instances of _rewards.microsoft.com_ to _rewards.bing.com_?

My searches are working fine, it's just the Rewards Tasks. Broke my 160 day streak!

**Comment by h4x0rm1k3** [comment]
@wearysky , I literally checked back on here 2 days ago to see if any issues and there was none outstanding so I literally missed you by a day! I only noticed today that my rewards hadn't done so glad you'd already picked up on it and worked out what was going on. I'm going to implement the fixes now and  upload them. Thanks for finding and bringing awareness.

**Comment by wearysky** [comment]
> @wearysky , I literally checked back on here 2 days ago to see if any issues and there was none outstanding so I literally missed you by a day! I only noticed today that my rewards hadn't done so glad you'd already picked up on it and worked out what was going on. I'm going to implement the fixes now and upload them. Thanks for finding and bringing awareness.

All good! The funny thing is, I did a Windows reinstall the day that it switched over, so I thought I had messed something up in my reinstall, somehow, and could NOT figure it out. It was only when I decided to take a closer look at the extension that I realized the problem.

Also: I started the v3 manifest conversion. The final issue I have to fix is that it keeps throwing inline code CSP errors for the element.click events, and there appears to be workarounds for that but I don't quite have my head wrapped around them yet.

**Comment by Filaipus** [comment]
I might have done something wrong but I noticed my rewards stopped doing the normal thing. Installed the new version from the latest commits. Anything else I need to do? It's still not working on opening up the links automatically from the rewards page.

**Comment by ChippewaBarr** [comment]
> I might have done something wrong but I noticed my rewards stopped doing the normal thing. Installed the new version from the latest commits. Anything else I need to do? It's still not working on opening up the links automatically from the rewards page.

Are you waiting a bit for the tasks to start?

I believe @h4x0rm1k3 implemented a 3 minute lag between the searches happening and the tasks starting. If I recall it was done since sometimes the tasks wouldn't complete it the searches were still taking place. Please someone correct me if I'm wrong.

**Comment by Filaipus** [comment]
> > I might have done something wrong but I noticed my rewards stopped doing the normal thing. Installed the new version from the latest commits. Anything else I need to do? It's still not working on opening up the links automatically from the rewards page.
> 
> Are you waiting a bit for the tasks to start?
> 
> I believe @h4x0rm1k3 implemented a 3 minute lag between the searches happening and the tasks starting. If I recall it was done since sometimes the tasks wouldn't complete it the searches were still taking place. Please someone correct me if I'm wrong.

I was trying to trigger them manually by pressing the "open rewards" button. Would i still need to wait in that use case?

**Comment by ChippewaBarr** [comment]
> > > I might have done something wrong but I noticed my rewards stopped doing the normal thing. Installed the new version from the latest commits. Anything else I need to do? It's still not working on opening up the links automatically from the rewards page.
> > 
> > 
> > Are you waiting a bit for the tasks to start?
> > I believe @h4x0rm1k3 implemented a 3 minute lag between the searches happening and the tasks starting. If I recall it was done since sometimes the tasks wouldn't complete it the searches were still taking place. Please someone correct me if I'm wrong.
> 
> I was trying to trigger them manually by pressing the "open rewards" button. Would i still need to wait in that use case?

Yes I believe so.

If I manually click that button nothing will happen for like 3 minutes then they will all open.

**Comment by h4x0rm1k3** [comment]
@Filaipus Yes you still have to wait, the only way to overide them is to open them each individually, I implemented a 3 minute delay a while back as the searches were interfering when rewards were going on.

**Comment by daedalus311** [comment]
Thanks for the update, brother!

**Comment by Filaipus** [comment]
Ok I waited the 3 minutes and it finally worked. I'm sorry for the impatience haha
Thanks for the great work, appreciate it!

**Comment by h4x0rm1k3** [comment]
> All good! The funny thing is, I did a Windows reinstall the day that it switched over, so I thought I had messed something up in my reinstall, somehow, and could NOT figure it out. It was only when I decided to take a closer look at the extension that I realized the problem.

Also: I started the v3 manifest conversion. The final issue I have to fix is that it keeps throwing inline code CSP errors for the element.click events, and there appears to be workarounds for that but I don't quite have my head wrapped around them yet.

@wearysky I just noticed the comment about manifest V3 and you're having some issues with CSP errors, any chance you mind sharing the code you've got so far and i'll take a look and see if I can puzzle out how to get it running, two heads and all that! I'm guessing it's only the rewards tasks/clicking that are the issue now or is search still an issue too in manifest v3? If you want to send it me directly you can email me at my username @gmail.com . Thanks

**Comment by wearysky** [comment]
@h4x0rm1k3 sent!

**Comment by DrDagless** [comment]
I updated the firefox add-on in the developer portal and it still doesn't seem to be opening for some reason. Have uninstalled and reinstalled a couple times now, but still not opening up after the initial searches complete (after 3 minutes).
